draft-ietf-quic-http.md

```diff
@@ -1117,7 +1117,55 @@ connection.  Clients do this by not sending additional requests on the
 connection; responses and pushed responses associated to previous requests will
 continue to completion.
 
-
+Servers initiate the shutdown of a connection by sending a GOAWAY frame
+({{frame-goaway}}).  The GOAWAY frame contains the Stream ID of the last
+client-initiated request that was or might be processed in this connection,
+which enables client and server to agree on which requests were accepted prior
+to the connection shutdown.  This identifier MAY be lower than the stream limit
+identified by a QUIC MAX_STREAM_ID frame, and MAY be zero if no requests were
+processed.  Servers SHOULD NOT increase the QUIC MAX_STREAM_ID limit after
+sending a GOAWAY frame.
+
+Once sent, the server MUST cancel requests sent on streams with an identifier
+higher than the indicated last Stream ID.  Clients MUST NOT send new requests on
+the connection after receiving GOAWAY, although requests might already be in
+transit. A new connection can be established for new requests.
+
+If the client has sent requests on streams with a higher Stream ID than
+indicated in the GOAWAY frame, those requests are considered cancelled
+({{request-cancellation}}).  Clients SHOULD reset any streams above this ID with
+the error code HTTP_REQUEST_CANCELLED.  Servers MAY also cancel requests on
+streams below the indicated ID if these requests were not processed.
+
+Requests on Stream IDs less than or equal to the Stream ID in the GOAWAY frame
+might have been processed; their status cannot be known until they are completed
+successfully, reset individually, or the connection terminates.
+
+Servers SHOULD send a GOAWAY frame when the closing of a connection is known
+in advance, even if the advance notice is small, so that the remote peer can
+know whether a stream has been partially processed or not.  For example, if an
+HTTP client sends a POST at the same time that a server closes a QUIC
+connection, the client cannot know if the server started to process that POST
+request if the server does not send a GOAWAY frame to indicate what streams it
+might have acted on.
+
+An endpoint MAY send multiple GOAWAY frames if circumstances change. For
```

Comment:
I know this commit is just moving text around, but I wanted to note how the beginning of this paragraph seems to be for the most part the same as in HTTP2 and does not take into consideration that in HQ there is no error code in the GOAWAY frame. So perhaps the "sends GOAWAY without an error code" can be removed ?

In HQ the only reason I can think of for the endpoint to send multiple GOAWAY frames is to first signal the intention to drain the connection, so that the remote endpoint on receipt stops creating new streams, and then after 1RTT (or on confirmation that the GOAWAY was received) the local endpoint can actually stop accepting new streams and seal the last stream ID and send a new GOAWAY frame. The remote endpoint upon receipt can now reset streams it thinks are established but in truth are not, with HTTP_REQUEST_CANCELLED, as explained above.